<!-- 
	SETUP.LAUNCH
	Multi-agent Trajectory Planning Simulation 

	USAGE : (at terminal) 1.roslaunch setup.launch
                              2.roslaunch runsim.launch
                (at matlab)   3.runsim
                
                If you want to try a new world, launch 'create_octomap_world.launch' to create new bt file
       
	Jungwon Park
	Seoul National Univ. 
	Lab for Autonomous Robotics Research
	wjstk335@gmail.com

-->
<launch>
  <!-- Parameters -->
  <arg name="world_name"   default="walls2"/>
  <arg name="world_x_size" default="12"/>
  <arg name="world_y_size" default="12"/>
  <arg name="world_z_size" default="2"/>

  <arg name="ecbs_w"        default="1.3"/>
  <arg name="ecbs_xy_res"   default="1"/>
  <arg name="ecbs_z_res"    default="1.5"/>
  <arg name="ecbs_margin"   default="0.3"/>

  <arg name="box_xy_res"    default="0.1"/>
  <arg name="box_z_res"     default="0.1"/>
  <arg name="box_margin"    default="0.4"/>

  <arg name="init_poses" default="0 -3 1 0 3 1 0.4
                                  1 -3 1 1 3 1 0.4
                                  2 -3 1 2 3 1 0.4
                                  3 -3 1 3 3 1 0.4
                                  -1 -3 1 -1 3 1 0.4
                                  -2 -3 1 -2 3 1 0.4
                                  -3 -3 1 -3 3 1 0.4
                                  -4 -3 1 -4 3 1 0.4"/>  
  <!-- Parameters End -->

  <!-- Nodes -->
  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server" args="$(find swarm_planner)/worlds/$(arg world_name).bt">
    <param name="resolution"   value="0.1" />
    <param name="frame_id"     value="world" type="string"/>
    <param name="sensor_model/max_range" value="5.0" />
  </node>

  <node pkg="swarm_planner" type="init_traj_generator" name="init_traj_generator" args="$(arg init_poses)" output="screen">
    <param name="world/x_size" value="$(arg world_x_size)" />
    <param name="world/y_size" value="$(arg world_y_size)" />
    <param name="world/z_size" value="$(arg world_z_size)" />

    <param name="ecbs/w"       value="$(arg ecbs_w)" />
    <param name="ecbs/xy_res"  value="$(arg ecbs_xy_res)" />
    <param name="ecbs/z_res"   value="$(arg ecbs_z_res)" />
    <param name="ecbs/margin"  value="$(arg ecbs_margin)" />

    <param name="box/xy_res"   value="$(arg box_xy_res)" />
    <param name="box/z_res"    value="$(arg box_z_res)" />
    <param name="box/margin"   value="$(arg box_margin)" />
  </node>
  <!-- Nodes End-->
</launch>
